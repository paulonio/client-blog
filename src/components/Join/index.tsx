import React from 'react';
import Link from 'next/link';
import Button from '../Button';
import { sen } from '@/styles/fonts';

import styles from './styled.module.scss';

const Join = () => {
  return (
    <section className={styles.join}>
      <div className={styles.wrapper}>
        <h2 className={sen.className}>Join our team to be a part of our story</h2>
        <p className="body-s">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.
        </p>
        <Link href="/contact">
          <Button>Join Now</Button>
        </Link>
      </div>
    </section>
  );
};

export default Join;
