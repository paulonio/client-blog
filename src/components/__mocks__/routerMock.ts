import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });
