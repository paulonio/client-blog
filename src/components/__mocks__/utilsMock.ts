import { isLinkActive, parseString } from '@/utils/utils';

jest.mock('../../utils/utils', () => ({
  isLinkActive: jest.fn(),
  parseString: jest.fn(),
}));

(isLinkActive as jest.Mock).mockReturnValue(true);
(parseString as jest.Mock).mockReturnValue('');
