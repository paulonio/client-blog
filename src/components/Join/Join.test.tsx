import React from 'react';
import { render } from '@testing-library/react';
import Join from '.';
import '../__mocks__/intersectionObserverMock';

describe('Join', () => {
  it('should render join', () => {
    const { container } = render(<Join />);

    expect(container).toMatchSnapshot();
  });
});
