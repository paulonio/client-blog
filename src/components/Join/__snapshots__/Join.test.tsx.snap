// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join should render join 1`] = `
<div>
  <section
    class="join"
  />
</div>
`;
